pkg/agent/agent.go

```diff
@@ -130,6 +130,7 @@ type Initializer struct {
 	enableL7FlowExporter  bool
 	connectUplinkToBridge bool
 	enableAntreaProxy     bool
+	enableMulticast       bool
```

Comment:
what there an issue with doing the check in `mcastController.Initialize` to avoid adding this field?

Reply:
I used to think about placing the check at the step before constructing multicastController instance, then some initial configurations may already be performed at that time, like OpenFlow, routes, which seems obey the thought that block at the beginning when configurations is not ready.

Reply:
I agree with @antoninbas. Other than the validation performed in startup stage, other validation logic should be put in its own module to better encapsulate the specific logic, instead of spreading the logic among multiple places, each of which does a piece of the whole. 

Generically, we should avoid making a "common" module include unrelated responsibilies. I'm not sure if AgentInitializer's responsibity is really clear today, but my understanding and expectation is that it is a module that initializes the bridge and the host network, not a validator for other modules.

Think about which is easier to maintain when doing something around Multicast: 
* add the static config handling in option.go, add runtime config validation in AgentInitializer, then add initialization error handling in Multicast controller
* add the static config handling in option.go, add other handling in Multicast controller

And consider what if a config can only be validated after you have made some system calls or API calls, it would be bad to move such business specific system calls or API calls to something like AgentInitializer.

Lastly, it's less redundant to have the validation in multicast controller itself because it doesn't have to check again whether the feature is enabled when doing the validation (the validation doesn't need to run when it's disabled).

Reply:
Got it, I will update accordingly.